ebpf/event_common.go

```diff
@@ -133,3 +137,27 @@ func (c ConnectionStats) ByteKey(buffer *bytes.Buffer) ([]byte, error) {
 	}
 	return buffer.Bytes(), nil
 }
+
+func max(a, b int) int {
+	if a > b {
+		return a
+	}
+	return b
+}
+
+func applyFuncOnRange(cs []ConnectionStats, isSame func(a, b *ConnectionStats) bool, f func([]ConnectionStats)) {
```

Comment:
While this looks kind of gnarly - it actually removes a whole lot of duplication + is easily testable :)